### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by TornaxO7**
Adjusted all tests so all are passing now.

# Question
While I was refactoring the tests, I encountered often pretty duplicated code like in the doc example of `config/model.rs` for the `new_with_envelope` method... I was thinking to create a new trait, which is only active during tests which you can use to create instances of a struct with real-case values, which makes it different compared to the `Default` trait.

In summary:
The `Test` trait should make it possible, to create new instances of a struct with near real-case attributes for testing while the `Default` trait can include "empty" attributes (like an empty vector or `None` and so on). What do you think?

### CODE REVIEW

**Comment by TornaxO7** [comment]
@soywod review pls

**Review by soywod — APPROVED** [review]

**Comment by TornaxO7** [comment]
@soywod could you also please answer my suggestion with the `Test` trait?

**Comment by soywod** [comment]
Mmh I'm not so fan of having testing code _outside_ of tests modules. It's not sth I've seen in other libs. I think we should not mix up codes. A dedicated function or a macro inside tests modules sounds more straightforward to me (and a better solution).

**Comment by TornaxO7** [comment]
> Mmh I'm not so fan of having testing code outside of tests modules.

Hold on, you're missunderstanding something.
I meant something like this:
`lib.rs`:
```rust
#[cfg(tests)]
pub trait Test {
    fn get_test() -> Self;
}
```

and then you can do the following for example for `msg/model/msg.rs`:
```rust
#[cfg(tests)]
impl Test for Msg {
    fn test() -> Self {
       // normal stuff
    }
}```

    }

**Comment by soywod** [comment]
What are the benefits of such a trait? I mean, we could just write tests directly inside the test module, why bothering with implementing a trait? Sorry if I don't understand well what you mean.

**Comment by TornaxO7** [comment]
Because I think that it would safe us some duplicated code. Let's say you need a `Msg` struct for a test in the imap-connection directory and another instance in the `body.rs` file. You'd create in both places [a similiar]/[an equal] function for both test modules (I **think**) or do you have a better idea? Because I'd like to avoid cases like [here](https://github.com/TornaxO7/himalaya/blob/mail_refactor/src/config/model.rs#L147) but that's probably my fault.

**Comment by soywod** [comment]
The `Msg` should be tested once at one particular place, we should not need to have multiple instances of `Msg`. Now, if you really see the use case then do, I'm following you!

**Comment by TornaxO7** [comment]
No wait, I don't want to force you!!! I just saw that you also used an account in the test for `imap` (it says that you don't use it) so I thought that it might suit if we include this trait.

**Comment by soywod** [comment]
No no you don't force me, I don't see the need but your are the one refactoring so you should know better, I really trust you! If you think it is needed then you can go, really :smiley:

**Comment by TornaxO7** [comment]
hm... ok, I'll think about it how to fix this...
